src/promptflow/promptflow/exceptions.py

```diff
@@ -209,3 +217,100 @@ class ValidationException(UserErrorException):
     """Exception raised when validation fails."""
 
     pass
+
+
+class ErrorInfo:
+    _exception_codes = dict()
+
+    @classmethod
+    def get_error_info(cls, e: Exception):
+        if not isinstance(e, Exception):
+            return None, None, None, None
+
+        e = cls.select_exception(e)
+        if cls._is_system_error(e):
+            return ErrorCategory.SystemError, cls._error_type(e), cls._error_target(e), cls._error_message(e)
+
+        return ErrorCategory.UserError, cls._error_type(e), cls._error_target(e), cls._error_message(e)
+
+    @classmethod
+    def select_exception(cls, e: Exception):
+        """Select the exception  in e and e.__cause__, and prioritize the Exception defined in the SDK."""
+
+        if e.__cause__ and isinstance(e.__cause__, PromptflowException):
+            return e.__cause__
+
+        if isinstance(e, PromptflowException):
+            return e
+
+        if e.__cause__ and isinstance(e.__cause__, HttpResponseError):
+            return e.__cause__
+
+        return e
+
+    @classmethod
+    def _is_system_error(cls, e: Exception):
+        if isinstance(e, (SystemErrorException, HttpResponseError)):
+            return True
+        if hasattr(e, "status_code") or (hasattr(e, "response") and hasattr(e.response, "status_code")):
+            status_code = str(e.status_code) if hasattr(e, "status_code") else str(e.response.status_code)
+            if not status_code.startswith("40"):
```

Comment:
4xx error like 429 should be usererror?

Reply:
I'm not sure if we should break down the status code. I refer to V1.5 SDK and it's determined directly based on 40.

Reply:
Except for 429, 400-499 are all client errors.